dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs

```diff
@@ -444,9 +438,15 @@ private async Task<string> GetCompletionAsync(IAIService aiService, ChatHistory
         throw new SKException("No AIService available for getting completions.");
     }
 
-    private int GetChatHistoryTokens(ChatHistory chatHistory)
+    private int GetChatHistoryTokens(ChatHistory chatHistory, string? additionalMessage = null, int skipStart = 0, int skipCount = 0)
     {
-        var messages = string.Join("\n", chatHistory.Messages);
+        var messages = string.Join("\n", chatHistory.Where((m, i) => i < skipStart || i >= skipStart + skipCount).Select(m => m.Content));
+
+        if (!string.IsNullOrEmpty(additionalMessage))
```

Comment:
Is there any reason why the GP3Tokenizer is not being used in here.

My suggestion would also to move this method to be a ChatHistory extension.